Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -497,6 +528,73 @@ float Read_F32(const u32 address)
   return Common::BitCast<float>(integral);
 }
 
+template <typename T>
+static TryReadResult<T> HostTryReadUX(const u32 address, RequestedAddressSpace space)
+{
+  if (!HostIsRAMAddress(address, space))
+    return TryReadResult<T>();
+
+  switch (space)
+  {
+  case RequestedAddressSpace::Effective:
+  {
+    const T value = ReadFromHardware<XCheckTLBFlag::NoException, T>(address);
```

Comment:
In order to handle the string cases more efficiently, since `const` can't be moved, I'd suggest:

```
T value = ReadFromHardware<XCheckTLBFlag::NoException, T>(address);
return TryReadResult<T>(!!MSR.DR, std::move(value));
```

(and the other variations below)